filecoin-proofs/tests/api.rs

```diff
@@ -1512,6 +1606,221 @@ fn create_seal_for_aggregation<R: Rng, Tree: 'static + MerkleTreeTrait>(
     )
 }
 
+fn create_seal_for_upgrade<R: Rng, Tree: 'static + MerkleTreeTrait<Hasher = TreeRHasher>>(
+    rng: &mut R,
+    sector_size: u64,
+    prover_id: ProverId,
+    porep_id: &[u8; 32],
+    api_version: ApiVersion,
+) -> Result<(SectorId, NamedTempFile, Commitment, TempDir)> {
+    init_logger();
+
+    let (mut piece_file, _piece_bytes) = generate_piece_file(sector_size)?;
```

Comment:
I believe we have some specific functionality to create emtpy sectors, probably should also have a test where the seal is created using that.